components/merkledb/src/views/metadata.rs

```diff
@@ -339,11 +375,78 @@ impl<T: RawAccess> IndexesPool<T> {
     }
 }
 
+impl<T: RawAccessMut> IndexesPool<T> {
+    /// Moves indexes with the specified prefix from the next version (i.e., `^prefix.*` form)
+    /// to the current version (`prefix.*` form). The existing old indexes are replaced, or
+    /// removed if the new index is a `Tombstone`. If there is no overriding index, an old
+    /// index is left in place.
+    ///
+    /// # Return value
+    ///
+    /// Returns resolved addresses of the removed indexes. For each address, we also return a flag
+    /// indicating whether the corresponding index was removed from aggregation (i.e., was
+    /// aggregated and was not replaced by an aggregated index).
+    pub(crate) fn flush_migration(&mut self, prefix: &str) -> Vec<(ResolvedAddress, bool)> {
+        let prefix = format!("^{}.", prefix);
+
+        // TODO: Collecting `moved_indexes` may be inefficient in terms of memory use.
+        let moved_indexes: Vec<_> = self.0.iter::<_, Vec<u8>, IndexMetadata>(&prefix).collect();
+        let mut removed_addrs = Vec::new();
+        for (key, metadata) in moved_indexes {
+            // `key[1..]` removes `^` char at the beginning of the index.
+            let migrated_key = &key[1..];
+            debug_assert!(migrated_key.starts_with(prefix[1..].as_bytes()));
+
+            if let Some(old_metadata) = self.0.get::<_, IndexMetadata>(migrated_key) {
+                let (resolved, is_in_group) =
+                    old_metadata.resolve_address(migrated_key, prefix.len() - 1);
+                let is_removed_from_aggregation = !is_in_group
```

Comment:
What exactly? `is_removed_from_aggregation` calculation? It's three lines of code, which are quite readable, IMO.

Reply:
I understand that appealing to authority is rarely a good thing, but can't avoid citing grandpa McConnell (:

>The single most import ant reason to create a routine is to reduce a program’s complexity. Create a routine to hide information so that you won’t need to think about it.

_Code Complete, chapter 7.1_

Here, IMO, it isn't important how exactly we check that index is removed from aggregation; and the method isn't straightforward itself, so if we can reduce complexity a bit, I think it's better to do so.

This is a matter of taste though, so you can freely ignore this comment :)